File: android/src/main/kotlin/com/sdk2009/sdk2009/receiver/AnyEventHandler.kt
package com.sdk2009.sdk2009.receiver

import io.flutter.plugin.common.EventChannel;

object AnyEventHandler : EventChannel.StreamHandler {

    private var eventSink: EventChannel.EventSink? = null
    override fun onListen(arguments: Any?, events: EventChannel.EventSink) {
        this.eventSink = events
        // Trigger events from native code when needed
        // Example: this.eventSink.success("Event data");
    }

    override fun onCancel(arguments: Any?) {
        this.eventSink = null
    }

    // Method to trigger the event from the plugin
    fun triggerEvent(eventData: String?) {
        eventSink?.success(eventData) // Send event data to Flutter
    }
}

File: ios/Classes/receiver/AnyEventHandler.swift
import Flutter
import UIKit
import Foundation

public class AnyEventHandler: NSObject, FlutterPlugin, FlutterStreamHandler {
    private var eventSink: FlutterEventSink?

    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        // Example: self.eventSink?("Event data")
        return nil
    }

    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.eventSink = nil
        return nil
    }

    // Method to trigger the event from the plugin
    public func triggerEvent(eventData: String) {
        if let eventSink = self.eventSink {
            eventSink(eventData) // Send event data to Flutter
        }
    }
}
